doc/confidence.rst

```diff
@@ -106,6 +106,167 @@ parameters and set it manually:
+
+The estimated values for the :math:`1-\sigma` standard error calculated by
+default for each fit includes the effects of correlation between pairs of
+variables, but assumes the uncertainties are symmetric.  While it doesn't
+exactly say what the values of the :math:`n-\sigma` uncertainties would be, the
+implication is that the :math:`n-\sigma` error is simply :math:`n^2\sigma`.
+
+
+The :func:`conf_interval` function described above improves on these
+automatically (and quickly) calculated uncertainies by explicitly finding
+:math:`n-\sigma` confidence levels in both directions -- it does not assume
+that the uncertainties are symmetric.  This function also takes into account the
+correlations between pairs of variables, but it does not convey this
+information very well.
+
+
+For even further exploration of the confidence levels of parameter values, it
+can be useful to calculation maps of :math:`\chi^2` values for pairs of
+variables around their best fit values and visualize these as contour plots.
+Typically, pairs of variables will have elliptical contours of constant
+:math:`n-\sigma` level, with highly-correlated pairs of variables having high
+ratios of major and minor axes.
+
+The :func:`conf_interval2d` can calculate 2-d arrays or maps of either
+probability or :math:`\delta \chi^2 = \chi^2 - \chi_{\mathrm{best}}^2` for any
+pair of variables.  Visualizing these can help better understand the nature of
+the uncertainties and correlations between parameters. To illustrate this,
+we'll start with an example fit to data that we deliberately add components not
+accounted for in the model, and with slightly non-Gaussian noise -- a
+constructed but "real-world" example:
+
+.. jupyter-execute::
+
+    # <examples/doc_confidence_chi2_map.py>
+    import numpy as np
+    import matplotlib.pyplot as plt
+    from lmfit import conf_interval, conf_interval2d, report_ci
+    from lmfit.models import GaussianModel, LinearModel
+    from lmfit.lineshapes import gaussian
+
+    rng = np.random.default_rng(seed=102)
+
+    #########################
+    # set up data -- deliberately adding imperfections and
+    # a small amount of non-Gaussian noise
+    npts = 501
+    x = np.linspace(1, 100, num=npts)
+    noise = rng.normal(scale=0.3, size=npts) + 0.2*rng.f(3, 9, size=npts)
+    y = (gaussian(x, amplitude=83, center=47., sigma=5.)
+         + 0.02*x + 4 + 0.25*np.cos((x-20)/8.0) + noise)
+
+    mod = GaussianModel() + LinearModel()
+    params = mod.make_params(amplitude=100, center=50, sigma=5,
+                             slope=0, intecept=2)
+    out = mod.fit(y, params, x=x)
+    print(out.fit_report())
+
+    #########################
+    # run conf_intervale, print report
+    sigma_levels = [1, 2, 3]
+    ci = conf_interval(out, out, sigmas=sigma_levels)
+
+    print("## Confidence Report:")
+    report_ci(ci)
+
+The reports show that we a pretty good fit, and that the automated estimates of
```

Comment:
"..... we OBTAINED a pretty good fit..."

Reply:
+1